src/server/bg_services/log_replication_scanner.js

```diff
@@ -0,0 +1,154 @@
+                if (!src_bucket || !dst_bucket) {
+                    dbg.error('log_replication_scanner: can not find src_bucket or dst_bucket object', src_bucket, dst_bucket);
+                    return;
+                }
+
+                const candidates = await log_parser.get_log_candidates(src_bucket._id, repl);
+
+                const keys_sizes_map_to_copy = await this.head_buckets_and_compare(src_bucket.name, dst_bucket.name, candidates);
+
+                dbg.log1('log_replication_scanner: keys_sizes_map_to_copy: ', keys_sizes_map_to_copy);
+                let move_res;
+                const keys_to_copy = Object.keys(keys_sizes_map_to_copy);
+                if (keys_to_copy.length) {
+                    const copy_type = replication_utils.get_copy_type();
+                    move_res = await replication_utils.move_objects(this._scanner_sem, this.client, copy_type, src_bucket.name,
+                        dst_bucket.name, keys_to_copy);
+                    console.log('replication_scanner: scan move_res: ', move_res);
+                }
+
+                await replication_store.update_log_replication_status_by_id(replication_id, Date.now());
+            });
+        }));
+    }
+
+    async head_buckets_and_compare(src_bucket, dst_bucket, candidates) {
+        const keys_sizes_map_to_copy = {};
+
+        // edge case 1: candidates = [] , nothing to replicate
+        if (!candidates.Contents.length) return keys_sizes_map_to_copy;
+
+        // head candidates in src_bucket and dst_bucket
+        const src_dst_objects_list = await this.head_objects(src_bucket, dst_bucket, candidates);
+
+        // edge case 1: src_head_map = {} , nothing to replicate
+        if (!src_dst_objects_list.length) return keys_sizes_map_to_copy;
+
+        for (let candidate of src_dst_objects_list) {
+
+            // case 1: object not found in src
+            if (!candidate.src_object_info) {
+                continue;
+            }
+
+            // case 2: object not found in dst
+            if (!candidate.dst_object_info) {
+                keys_sizes_map_to_copy[candidate.key] = candidate.ContentLength;
+                continue;
+            }
+
+            // case 3: object present both on the src and dst, need to check_data_or_md_change
+            const should_copy = replication_utils.check_data_or_md_changed(candidate.src_object_info, candidate.dst_object_info);
+            if (should_copy) {
+                keys_sizes_map_to_copy[candidate.key] = candidate.ContentLength;
+            }
+        }
+        return keys_sizes_map_to_copy;
+    }
+
+    async head_objects(src_bucket_name, dst_bucket_name, candidates) {
+        const src_dst_objects_info = await P.all(_.map(candidates, async key => {
+            try {
+                dbg.log1('log_replication_scanner head_object: params:', src_bucket_name, key);
+
+                const src_dst_object_info = await P.all([await this.noobaa_connection.headObject({
```

Comment:
this array of 2 items originates from an old version of this function? I don't see why it's needed